.gitlab-ci.yml

```diff
@@ -93,17 +93,6 @@ default:
     - docker
     - linux
 
-min_boost:
```

Comment:
why do you want to remove this test?

Reply:
Here's how I understood our last es meeting:
* we don't test boost 1.55 any more
* boost 1.58 is already tested in `cuda:9.0` and `clang:6.0`
* once Ubuntu 20.04 LTS is released next April, we upgrade boost requirements to 1.65 (for `cuda:9.0` which requires Ubuntu 16.04, we build boost 1.65 from sources)
*  except for building boost from sources, `Dockerfile-min_boost` = `Dockerfile-18.04`, so we can transfer it from master to a new 4.1 branch

Reply:
We could think about having one container where all the library versions are pinned to the minimal required versions, so that we don't silently stop testing this when a library version is updated in the distribution. This isn't really complicated, all major package managers let you just fix a version.

Reply:
`cuda:9.0` will essentially fulfill that role. It uses Ubuntu 16.04 and the minimal supported versions of CUDA and CMake. The version numbers of non-graphical Python packages available in the Ubuntu 16.04 repository are used as the minimal versions in `requirements.txt` (fb4862ed894e7). Once Ubuntu 20.04 is out and espresso drops support for CUDA9, we can use the `cuda:10.1` to fulfill that role. We could rename the CI job from `cuda10:maxset` to `ubuntu:min-dependencies` for clarity.

Reply:
Whatever you think is appropriate. I'm just saying that it is not as such guaranteed that the package versions in the distributions stay the same, which could lead to a situation where the minimal version of some package is no longer tested.

Reply:
We could come up with an extra check in the CI job that tests the docker image to make sure the versions of dependencies match the minimal requirements.